subiquity/models/filesystem.py

```diff
@@ -1444,13 +1457,29 @@ def can_emit(obj):
                 raise Exception("\n".join(msg))
             work = next_work
 
+        if early:
+            r = [act for act in r if act['type'] not in ('format', 'mount')]
+        if late:
+            r = [act for act in r if act['type'] in ('format', 'mount')]
```

Comment:
The connection between early & late & mounts is not immediately clear, I suggest a comment.
if early and later are both true, we'll drop the list from r, which I assume is always an error but would go unnoticed.

Reply:
I did a bit of refactoring here to make this all a bit clearer (basically _render_actions now takes an enum).

Reply:
Thank you, this leaves less room for surprises for the caller.